apps/smtp/src/modules/smtp/configuration/smtp-metadata-manager.test.ts

```diff
@@ -0,0 +1,43 @@
+import { SettingsManager } from "@saleor/app-sdk/settings-manager";
+import { beforeEach, describe, expect, it, vi } from "vitest";
+
+import { SmtpMetadataManager } from "./smtp-metadata-manager";
+
+const SALEOR_API_URL = "https://saleor-api.com";
+
+class MockSettingsManager implements SettingsManager {
+  get = vi.fn();
+  set = vi.fn();
+  delete = vi.fn();
+}
+
+describe("SmtpMetadataManager", () => {
+  beforeEach(() => {});
+
+  describe("getConfig", () => {
+    it("returns configuration without errors", async () => {
+      const metadataManager = new MockSettingsManager();
+
+      vi.spyOn(metadataManager, "get").mockResolvedValue('{"key": "value"}');
+
+      const instance = new SmtpMetadataManager(metadataManager, SALEOR_API_URL, 1000);
+      const result = await instance.getConfig().unwrapOr("error");
+
+      expect(result).toEqual({ key: "value" });
+    });
+
+    it("raises error when pulling configuration exceeds the timeout", async () => {
+      const metadataManager = new MockSettingsManager();
+
+      vi.spyOn(metadataManager, "get").mockImplementation(
+        () => new Promise((resolve) => setTimeout(resolve, 100)),
```

Comment:
probably you should use something like 
https://vitest.dev/api/vi.html#vi-advancetimersbytime

and setTimeout can be something more realistic like 15s, yet fake timers can make it testable